ui/segmented-control.mod/segmented-control.js

```diff
@@ -134,24 +150,22 @@ const SegmentedControl = (exports.SegmentedControl = class SegmentedControl exte
                 // If no segment is selected, hide the thumb
                 this.thumbElement.style.display = "none";
             }
-
-            if (!this._readyForAnimation && !this._shouldEnableAnimation) {
-                // Trigger a redraw to enable animations
-                this._shouldEnableAnimation = true;
-                this.needsDraw = true;
-                return;
-            }
         } else {
             // Wait for the inner repetition segments to be drawn, before moving the thumb
             segments.addEventListener("firstDraw", this.handleSegmentsFirstDraw, false);
         }
+    }
 
-        if (this._shouldEnableAnimation && !this._readyForAnimation) {
-            // Enable animations after the initial positioning, to avoid unwanted transitions
-            this.element?.classList.add("mod--readyForAnimation");
-            this._readyForAnimation = true;
+    /**
+     * Handles the end of thumb transition animations
+     */
+    handleTransitionEnd = () => {
```

Comment:
handleTransitionEnd has one argument that is an event of a known type, please add it to respect the signature of the method, even if it's not used within the method